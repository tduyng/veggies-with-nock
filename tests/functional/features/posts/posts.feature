@posts
Feature: Get posts

    Scenario: I can create a new post
        Given I mock a POST http call to forward request body for path /posts
        And set request json body from create_post
        When I POST http://mysite.com/api/v1/posts
        Then response status code should be 200
        And response json body should match snapshot
            | field       | matcher | value        |
            | post.userId | equal   | 10((number)) |
            | post.id     | equals  | 1((number))  |

    Scenario: I get a list of posts
        Given I mock a GET http call to forward request body for path /posts
        When I GET http://mysite.com/api/v1/posts
        Then response status code should be 200
