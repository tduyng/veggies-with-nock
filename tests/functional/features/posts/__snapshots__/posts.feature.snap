

exports[`I can create a new post 1.1`] = `Object {
  "post": Object {
    "id": "equals(1((number)))",
    "text": "quia et suscipit
suscipit recusandae consequuntur expedita et cum
reprehenderit molestiae ut ut quas totam
nostrum rerum est autem sunt rem eveniet architecto",
    "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
    "userId": "equal(10((number)))",
  },
}`;
